rfcs/039-requesting-api-design/README.md

```diff
@@ -0,0 +1,215 @@
+        {
+          "locationType": {
+            "id": "closed-stores",
+            "label": "Closed stores",
+            "type": "LocationType"
+          },
+          "label": "Closed stores",
+          "shelfmark": "Pat. vol. 157:30",
+          "accessConditions": [],
+          "type": "PhysicalLocation"
+        }
+      ],
+      "status": {
+        "id" : "not-requestable",
+        "label" : "Not requestable",
+        "type": "ItemStatus"
+      },
+      "type": "Item"
+    },
+    {
+      "locations": [
+        {
+          "locationType": {
+            "id": "iiif-presentation",
+            "label": "IIIF Presentation API",
+            "type": "LocationType"
+          },
+          "url": "https://wellcomelibrary.org/iiif/b30758762/manifest",
+          "license": {
+            "id": "pdm",
+            "label": "Public Domain Mark",
+            "url": "https://creativecommons.org/share-your-work/public-domain/pdm/",
+            "type": "License"
+          },
+          "accessConditions": [
+            {
+              "status": {
+                "id": "open",
+                "label": "Open",
+                "type": "AccessStatus"
+              },
+              "type": "AccessCondition"
+            }
+          ],
+          "type": "DigitalLocation"
+        }
+      ],
+      "status": {
+        "id" : "not-requestable",
+        "label" : "Not requestable",
+        "type": "ItemStatus"
+      },
+      "type": "Item"
+    }
+  ],
+  "type": "Work",
+  "@context": "https://api.wellcomecollection.org/catalogue/v2/context.json"
+}
+```
+
+</details>
+
+
+### Managing requests on items
+
+We will add a new endpoint to `account.wellcomecollection.org`: `/users/{userId}/item-requests`.
+This goes alongside other requests on this endpoint, as listed [in the identity repo](https://github.com/wellcomecollection/identity/blob/0de73ee41a332eb24ee0b4e2d37bc95e0282255f/packages/apps/api-authorizer/src/index.ts#L164-L211).
+
+The three operations are as follows:
+
+-   To request an item, a user makes a POST request to this endpoint.
+
+    This POST request will include a body of the following format:
+
+    ```json
+    {
+      "patronId": "1234567",
+      "workId": "a23upgqy",
+      "itemId": "fwsbxqp4",
+      "type": "ItemRequest"
+    }
+    ```
+
+    This endpoint will return the following status codes:
+
+    -   201 Created if the request is successful (or already exists)
+    -   403 Forbidden if the user is unable to place requests (e.g. at their account limit, banned account)
+    -   404 Not Found if the work does not exist, or the work exists but the item is not part of this work
+    -   409 Conflict if the work is already requested by another user
+
+-   To retrieve a list of their pending requests, a user makes a GET request to this endpoint.
+
+    This will return a list of ItemRequest objects:
+
+    ```json
+    [
+      "requests": [
+        {
+          "id": "sierra/1234567",
```

Comment:
I'm going to drop this until we bring relinquishing requests back into scope.

I'm not sure if you can rely on GUIDs – it adds another data store and potential failure point. e.g. if we successfully record the hold in Sierra but don't manage to store our GUID, what do we show the user? The request didn't totally "succeed" (because they can't relinquish it), but nor did it "fail" (staff will be retrieving the item for them).

Reply:
> if we successfully record the hold in Sierra but don't manage to store our GUID
We would record the attempt to place a request before making one, then update with a confirmation that the request succeeded I think. But we shouldn't think about this yet (as you say).

Reply:
> I'm assuming the generally serialisation here is designed to be consistent with the rest of the identity API?

Yup.